### DISCUSSION TITLE
No Loras are working

### DISCUSSION THREAD

**Discussion by juliux32** [lllyasviel/Fooocus#1475] [Q&A]
I'm new to all this AI thing so maybe I am doing something wrong and would like to get some guidance...

So basically my problem is, that I have tried several Loras and none of them works... for example, I use the simple prompt "girl walking in park" and set the body weight Lora slider to 2 and nothing happens... slide it to -2 and still the same size person... tried Loras like hair length, breast size, skin tone, body weight, body muscularity, and no matter if I set any of those to -2 or 2, I always get basically the same person... I tried to write things in the prompt like <name_of_lora:2> or <lora:name_of_lora:2> but still nothing...

I download Loras from Civit.ai and simply copy/paste them to Fooocus_win64_2-1-831\Fooocus\models\loras so maybe I am doing something wrong here? Do I have to import them in some way?

Any advice, help, or teaching would be appreciated

**Comment by mashb1t** [comment]
For some LoRAs a trigger word in the prompt is needed after selecting it in the Model tab.

Example:
LoRA: https://civitai.com/models/206995/star-wars-characters-xl
Trigger word: Grogu

Please check your LoRA(s) for trigger words and use them in the prompt.

**Reply by juliux32** [reply]
And what if there are no trigger words?

**Reply by mashb1t** [reply]
LoRA's will nevertheless work up to an extend without trigger words.
Please make sure that you only use LoRAs which have explicitly been trained for SDXL as many of the ones you listed were created only for 1.5.

SD 1.5: https://civitai.com/models/114215

![Hair_Length_Slider_-_LoRA_-_v1_0___Stable_Diffusion_LoRA___Civitai](https://github.com/lllyasviel/Fooocus/assets/9307310/d6c3b11d-66ca-401c-ae85-e11ff72195e4)

SDXL (the one you need): https://civitai.com/models/180780

![Crystals__LoRA_1_5_SDXL__-_SDXL___Stable_Diffusion_LoRA___Civitai](https://github.com/lllyasviel/Fooocus/assets/9307310/4f445a07-282b-4cb7-a2a9-5117564ddabf)

**Reply by juliux32** [reply]
Oh... then it's a really big con... because there's basically no SDXL Loras... that is pitty because I started enjoying Fooocus

**Reply by lllyasviel** [reply]
download a 1.5 model as refiner and set switch to like 0.4 to use 1.5 loras

**Reply by mashb1t** [reply]
haven't thought about that but @juliux32 hope this workaround works for you, even when not being able to use the LoRAs all the way :)

**Reply by sinewave-2097** [reply]
I downloaded goofyai's disney_style safetensor into models/checkpoints and while fooocus does see it, selecting it as a refiner results in errors. It is an SDXL model, according to the readme: https://huggingface.co/goofyai/disney_style_xl/tree/main Is there something else I should be looking for that would be a trigger for a model not working?

KeyError: 'model.diffusion_model.input_blocks.0.0.weight'

**Reply by mashb1t** [reply]
@sinewave-2097 please put disney_style in the lora folder and select it as such, not in checkpoints. A common SD 1.5 model (checkpoint) you can use as a refiner is dreamshaper 8, https://civitai.com/models/4384

**Reply by sinewave-2097** [reply]
Thanks Mashb1t, I didn't make the distinction, but now that makes sense. It works for now, thank you again!

**Reply by juliux32** [reply]
@lllyasviel I put it as a refiner, but now I am getting this error when trying to generate
 **KeyError: 'model.diffusion_model.input_blocks.0.0.weight'**

**Reply by mashb1t** [reply]
the answer is literally 2 comments above 😅
https://github.com/lllyasviel/Fooocus/discussions/1475#discussioncomment-7880313

**Reply by juliux32** [reply]
@mashb1t So I don't understand @lllyasviel advice to use my Loras like hair length, breast size, skin tone, body weight, body muscularity as a refiner if I can't use them 🫤 I don't know what is your offered DreamShaper 8, but I am pretty sure it is not controling things like weight, hair etc 😕

**Reply by mashb1t** [reply]
This is not what was mentioned. You must not use LoRAs as refiner, but a SD 1.5 model such as dreamshaper.
Please move the LoRAs from the checkpoints to the lora folder, download a SD 1.5 model, put it in checkpoints and select it as refiner + add your LoRA in the LoRA section.

**Reply by lllyasviel** [reply]
@juliux32 if you are relatively new to these, you may stick to prompting and just use default settings. This is why we made fooocus for new people; usually prompting can solve most problems since the style system and SDXL is very stong.

if you use are new to civitai, the suggestion is to use exactly same software and setting as proposed by each civitai model author. Then after familiar with related concepts, you can move on to play with different software and settings.

But subjectively we are confident that user will come back to Fooocus because using other software to outperform Fooocus default quality is difficult, even with many models/settings from civitai (because if anything is really good enough then we will just set it as a preset).

**Reply by juliux32** [reply]
@lllyasviel I have tried Automatic1111 because everyone said that it is the most powerful tool, but I got terrible results... Maybe because I am new and don't know how to tweak things around, but even a simple prompt in 1111 "girl in the park", gave me messed up face girls and based on what I saw on youtube, you need to add a bunch of extra loras, extensions and etc., like face fix, quality upscaler, clear skin... Maybe if you are experienced in this, 1111 could give you better results than Fooocus, because there are a lot of extensions, plugins, and features, but that could be a way to steep learning curve if you need something simple...

**Reply by lllyasviel** [reply]
@juliux32 hi 
all software should give competitive results and then they compete in some way.
if AMC1111 give clearly wrong images, then it is most likely wrong user operations rather than software problem. For AMC1111 new users, the recommended way is to use exactly same setting as proposed by each civitai model author.

**Reply by datathehuman** [reply]
It doesn't help to accuse people of things; I don't think, juliux32 that you are following their kind instructions. Thank you's also help lubricate the social environment. Just slow down and be patient and read the instructions carefully. Have a nice day.

**Reply by mati-fernandez** [reply]
@juliux32 I got the same issue. I'm trying with a lora I trained with my face. It worked on automatic1111 but there the problems were others, like you mencioned, but at least it loads my lora properly with my keyword. In aut1111 when is "mixing, loading" images, it seems that the result will be great (photorealistic, no deformation) but at the end it changes to a same pic with a little bit of anime style and deformation. I revised all the settings, didn't change anything but I'm here trying with fooocus again because of that.

 UPDATE: I trained with oneTrainer two new models. One with SD 2.1 and the other with SDXL 1.0 Base (both type LoRa). I tried again with fooocus in the loras setting with these models and nothing happened :(

**Comment by mati-fernandez** [comment]
@juliux32 I got the same issue. I'm trying with a lora I trained with my face. It worked on automatic1111 but there the problems were others, like you mencioned, but at least it loads my lora properly with my keyword. In aut1111 when is "mixing, loading" images, it seems that the result will be great (photorealistic, no deformation) but at the end it changes to a same pic with a little bit of anime style and deformation. I revised all the settings, didn't change anything but I'm here trying with fooocus again because of that.
 
 UPDATE: I trained with oneTrainer two new models. One with SD 2.1 and the other with SDXL 1.0 Base (both type LoRa). I tried again with fooocus in the loras setting with these models and nothing happened :(

**Comment by RiccardoFragale** [comment]
As far as my understanding goes, if you are using Fooocus you are working with SDXL models, this means that you need XL Loras for your models. Are you trying to use SD1.5 Loras? Those don't work for me either. To find XL Loras on Civitai you can use the search filter and select all the SDXL model types.

Try with the following Lora:
https://civitai.com/models/251417/midjourney-mimic?modelVersionId=283697

On top of the following base model:
https://civitai.com/models/112902/dreamshaper-xl?modelVersionId=351306

with this prompt:"([by Mark Keathley|by Arnold Böcklin]:1.5), (The mighty Minotaur, trapped in a labyrinth, monstrous, and ferocious, awaits in the heart of a maze. Its fur is a deep brown, and its eyes are filled with an intense energy. The creature's body is covered in intricate patterns and symbols, and it seems to be waiting for a challengers to come forward. The scene is set in a vast, barren landscape, with a lone figure in a long coat and a thick mask preparing to take on any potential threats.:0.5) (Clockpunk fantasy:1.3) <lora:MJ52:0.4>"

you should get similar results to the following:

- without the Lora (the Minotaur is not inside a labyrinth or any kind of structure):
![image](https://github.com/lllyasviel/Fooocus/assets/14418179/89e12bcf-ed50-402e-80f1-bfcfa27bf90d)

- with the Lora (the Minotaur is inside a labyrinth or some sort of structure):
![image](https://github.com/lllyasviel/Fooocus/assets/14418179/c9a90291-6491-4c7e-ba16-d0ff6646da11)

enjoy!
